Controller/ContentSecurityPolicyController.php

```diff
@@ -66,7 +67,11 @@ public function indexAction(Request $request)
         }
 
         if ($this->dispatcher) {
-            $this->dispatcher->dispatch(Events::VIOLATION_REPORT, new Event($report));
+            if (Kernel::VERSION_ID >= 40300) {
```

Comment:
this must be detected based on the `symfony/event-dispatcher` version, not based on the `symfony/http-kernel` version. It is possible to have different versions of these packages installed.

The easiest check here would be to check if the dispatcher implements the new EventDispatcherInterface from contracts.